src/inc/RingBuffer.h

```diff
@@ -6,146 +6,150 @@
 #include <cmath>
 #include <cstring>
 
+// Helper function because C++'s modulo operator (%) can return negative results.
```

Comment:
This is very interesting, and good to know! Good catch here!

Reply:
Yeah, C/C++ has truncated modulo, which is unfortunately never what I want (since you have to take extra care in handling negative dividends).
(There's a nice summary of the variations on modulo [here](https://en.wikipedia.org/wiki/Modulo_operation), along with a table of which version different languages offer.)